src/main/java/org/checkerframework/languageserver/CFTextDocumentService.java

```diff
@@ -163,4 +171,34 @@ public void publish(Map<String, List<javax.tools.Diagnostic<?>>> result) {
                                     .collect(Collectors.toList())));
         }
     }
+
+    /**
+     * The hover request is sent from the client to the server to request hover information at a
+     * given text document position.
+     *
+     * <p>Registration Options: TextDocumentRegistrationOptions
+     */
+    @Override
+    public CompletableFuture<Hover> hover(HoverParams params) {
+        generateTestData();
+        // line 1 on vscode is 0
+        int line = params.getPosition().getLine();
```

Comment:
Both line and col numbers matter because you can have multiple identifiers on the same line.